### DISCUSSION TITLE
Priiloader 0.10.0 Beta 2  - Testing and Info - New Hacks, vWii Support and More!

### DISCUSSION THREAD

**Discussion by Ingunar** [DacoTaco/priiloader#336] [General]
Hey everyone, I created a new permanent link to the note.
We use this note to keep track of the current tests, bugs, and progress on Priiloader's development.

[**Priiloader 0.10.0_Beta2 Testing**](https://hackmd.io/@Ingunar/PriiloaderTesting)

**Also, we need some help with testing Priiloader hacks on JAP Wiis and JAP vWiis, and Wii Minis from all of the regions. 
You don't need to install a beta priiloader build to test the hacks! They are compatible with the stable release.
You can find more info about the hacks in the note.**

**Comment by Ingunar** [comment]
**Priiloader 0.10.0 beta2 - Stable Test - Known common issues and Info:**

__Known issues:__

- Gecko Dumping to SD in some very rare cases doesn't work. We didn't manage to replicate this.
(Very rarely can happen, If you cold boot and autoboot is set to SM)

__Wii System Settings on vWii - Additional info:__

- Currently, anything changed inside the System settings will be reset to the previous settings after the console's reboot.
Almost every setting can be changed permanently inside the WiiU's settings.
New hacks to resolve this are planned to be created in the future.
You can find full info in the Testing note's: "Explanation about the Wii System Settings and the permanent SYSCONF solution" section.

__Other Info:__

- ~~**BootMii** iOS doesn't exist for vWii~~ - It probably never will. This option can not be currently used with vWii
- ~~USB Loader GX currently has graphical glitches with the Channel Banners~~ - The developer fixed it, and the update for it will be released soon!
- **Using MyMenuify to change vWii System Menu theme will cause brick.** 
(The app will need an update to work with Priiloader, but can be used to change the theme before the Priiloader is installed)
- WiiConnect24 tasks are not confirmed to be fully working.

__Hacks support is still incomplete. Detailed information can be found in [the note](https://hackmd.io/@Ingunar/PriiloaderTesting)__

**Reply by DardelHMB** [reply]
Dolphin also needs to be updated to support vWii priiloader it seems. Attempting to load the system menu from a dumped vwii nand with priiloader installs results in it getting stuck in the priiloader menu with the message "Failed to load system menu!". Games work fine. On a real Wii (cannot confirm for beta2 version) this doesnt happen.

**Reply by Ingunar** [reply]
Yes, the current version of Dolphin does not fully support it.  It can't decrypt vWii System Menu without the `/dev/aes`

**Comment by Ingunar** [comment]
### **Priiloader 0.10.0.beta2 - Stable Test 02 (r0x219b3d5) is released**

Changes and fixes: 

-  Make the installer check that BC-NAND is actually installed properly and parse its TMD.
-  **Fixed:** Selecting non-HDMI in the Wii U settings causes the System Menu to display incorrectly.
- Add support for [boot args](https://wiiubrew.org/wiki/WiiMode#Boot_arguments) (to automatically launch a channel from Wii U mode).
- Prevent installing on old vWii SM versions.

**Comment by Ingunar** [comment]
### **Priiloader 0.10.0.beta2 - Stable Test 03 (r0x194c613) is released**

Changes and fixes: 

- JAP and USA VI patches

**Reply by qwertyuiopasd1113** [reply]
None of the hacks that i tried worked for me. I'm on a PAL Wii U, using Priiloader 0.10.0.beta2 Stable Test 03 (r0x194c613)
These are the ones I tried:
Replace Health Screen with Backmenu
Move Disc Channel
Enable Wii System Settings for vWii
Message board write for vWii
Always enable WiiConnect24 for vWii

**Reply by Ingunar** [reply]
Those hacks are all confirmed to be working.
 
- Did you use `hacks_hash.ini` from the .zip?
- Are those the only hacks that didn't work?
- Are those the only hacks that you tried to enable?
- Did you maybe forget to select `save settings` after enabling the hacks in the `System Menu Hacks` section?
 (That happens more often than you think...)

**Reply by qwertyuiopasd1113** [reply]
I did the "The latest version of hacks_hash.ini to test - Version 11.05.23" from the hackmd page
Those are the only hacks i tried, none of them worked. I pressed Save settings as it said settings saved successfully or whatever the message is.

**Reply by Ingunar** [reply]
Yes, those are currently the same ones as in the archive.

- If you enter priiloader menu now, are they enabled? Do they stay enabled?
- Did you try to reinstall the priiloader or hacks?
- Did you have any priiloader build installed before that one?

**Reply by DacoTaco** [reply]
also another thing: sharing the Prii.log could help, specially if we make custom builds to test

**Reply by qwertyuiopasd1113** [reply]
They stay enabled, I reinstalled both priiloader and hacks_hash.ini. I had no other priiloader builds.
Sorry for the late response :sweat_smile: 
I had to enable dumping gecko output to get prii.log, here it is:
[prii.log](https://github.com/DacoTaco/priiloader/files/11971396/prii.log)

**Reply by Ingunar** [reply]
Can you do 2-3 reboots, and then send the `prii.log` again?

also interesting
`20:06:46 : USB: Failed to mount`

**Reply by Ingunar** [reply]
Did you maybe edit out some hacks or edited `hacks_hash.ini` at some point?

If you could post your `hacks_hash.ini` from the priiloader folder, that could be useful.

**Reply by DacoTaco** [reply]
i was more interesting of the log of you booting SM, instead of you booting HBC but ok :)

**Reply by qwertyuiopasd1113** [reply]
[prii.log](https://github.com/DacoTaco/priiloader/files/11985830/prii.log)
[hacks_hash.ini](https://github.com/DacoTaco/priiloader/files/11985834/hacks_hash.ini.txt)
`USB: Failed to mount` might be because of my Wii U formatted hard drive

**Reply by Ingunar** [reply]
Hmm, could you please:
- Delete the the log from your SD card
- Disable [Always enable WiiConnect24 for vWii] and enable [Move Disc Channel] hack
- Save the change
- Boot to SM, and reboot 1-2 times to the vWii SM again after that
- Try moving the Disc channel to see if the hack enabled it
- Upload the newly generated `prii.log` here again

**Reply by qwertyuiopasd1113** [reply]
I also disabled all hacks other than Move Disc Channel, it didn't work
[prii.log](https://github.com/DacoTaco/priiloader/files/11986647/prii.log)

**Reply by Ingunar** [reply]
Well, don't touch anything. 🥲 We need to wait for @GaryOderNichts and @DacoTaco to see what should be the next step...

Had the same bug months ago, when we where porting hacks, but couldn't replicate it,
My current guess is that the [Always enable WiiConnect24 for vWii] is causing it somehow.

**Reply by DacoTaco** [reply]
@Gauley123 : i would like to have a copy of the hacks_hash.ini on your sd card (and usb?) AND a copy of `/title/00000001/00000002/data/hackshas.ini` on the (v)wii nand. i would also like a copy of `/title/00000001/00000002/data/hacksh_s.ini`.
this way i can hopefully investigate or reproduce the error.

**Reply by qwertyuiopasd1113** [reply]
Here:
[hacks_hash.ini](https://github.com/DacoTaco/priiloader/files/11995390/hacks_hash.ini.txt)
[hacksh_s.ini](https://github.com/DacoTaco/priiloader/files/11995391/hacksh_s.ini.txt)
[hackshas.ini](https://github.com/DacoTaco/priiloader/files/11995392/hackshas.ini.txt)

**Reply by Ingunar** [reply]
Thank you @Gauley123, you helped us a lot!
Daco managed to replicate it finally and fixed it.

I'll upload the Stable Test 04 soon. 
The hacks should start working after you do the update.

**Comment by Ingunar** [comment]
### **Priiloader 0.10.0.beta2 - Stable Test 04 (r0x69d1659) is released**

Changes and fixes: 

-  Hack cache bug fix
-  Added new official HBC image
-  Included [updated hacks](https://hackmd.io/@Ingunar/PriiloaderTesting#Version-History) 
- [Priiloader Forwarder for Aroma - WiiU](https://cdn.discordapp.com/attachments/1078641159510958090/1127656412072390777/PriiloaderWiiUForwarder_Aroma.zip) added
- [LoadPriiloader - HBC (Wii,vWii,Mini)](https://cdn.discordapp.com/attachments/1078641159510958090/1127656824565399662/LoadPriiloader_HBC_WiivWiiMini.zip) updated

**Comment by Ingunar** [comment]
### **IMPORTANT INFORMATION**


**It has been reported that if the Priiloader is installed, using MyMenuify to change vWii System Menu theme can cause brick!**

MyMenuify, same as USB Loader GX, needs to be updated to work with vWii Priiloader.
Be careful if you are using any app that changes/works with the Wii System Menu asset archive.
If you know any similar app that should be checked, tell me.

**Comment by qwertyuiopasd1113** [comment]
Could there be a hack to enable [sending a photo](https://www.nintendo.co.uk/Support/Wii/Usage/Wii-Message-Board/Sending-a-Photo/Sending-a-Photo-240112.html) from the Wii Message Board on vWii?

**Reply by Ingunar** [reply]
With the advanced hack maybe. We are still looking how to solve that, not the biggest priority atm.

**Comment by eku** [comment]
> ### **IMPORTANT INFORMATION**
> 
> **It has been reported that if the Priiloader is installed, using MyMenuify to change vWii System Menu theme can cause brick!**
> 
> MyMenuify, same as USB Loader GX, needs to be updated to work with vWii Priiloader. Be careful if you are using any app that changes/works with the Wii System Menu asset archive. If you know any similar app that should be checked, tell me.

Would be nice, if you could check [WiiFlow Lite](https://github.com/Fledge68/WiiFlow_Lite) as well.  Is also a USB loader like ULGX and could be affected as well.

**Reply by Ingunar** [reply]
WiiFlow Lite is not using SM Assets, so it is working properly. Only those 2 are known to have issues. 

USB Loader GX dev is working on the fix.

**Reply by eku** [reply]
Thanks

**Comment by Siiyar** [comment]
> ### **Priiloader 0.10.0.beta2 - Stable Test 04 (r0x69d1659) is released**
> Changes and fixes:
> 
> * Hack cache bug fix
> * Added new official HBC image
> * Included [updated hacks](https://hackmd.io/@Ingunar/PriiloaderTesting#Version-History)
> * [Priiloader Forwarder for Aroma - WiiU](https://cdn.discordapp.com/attachments/1078641159510958090/1127656412072390777/PriiloaderWiiUForwarder_Aroma.zip) added
> * [LoadPriiloader - HBC (Wii,vWii,Mini)](https://cdn.discordapp.com/attachments/1078641159510958090/1127656824565399662/LoadPriiloader_HBC_WiivWiiMini.zip) updated

Priiloader forwarder for aroma doesn't seem to work at all. When launched, it just reboots the wii u. I am on the newest version of aroma as on writing this. LoadPriiloader seems to work fine though.

**Reply by DacoTaco** [reply]
reboots to wiiu or vwii mode?

**Reply by Ingunar** [reply]
That happens when you don't have Priiloader installed.

**Reply by Siiyar** [reply]
I have Priiloader installed.

**Reply by Ingunar** [reply]
Are you using RC3?

**Reply by Siiyar** [reply]
Yeah

**Reply by Ingunar** [reply]
**Ok, I am going to need this info:**

- Where did you download the forwarder from?

- What Aroma version are you using?

- What region is your WiiU?

- Did you try reinstalling Priiloader and Priiloader forwader?

**Reply by Siiyar** [reply]
From the site you provided

Aroma Beta 16 (Latest)

European

Yep

**Reply by Siiyar** [reply]
Update: I reinstalled both of them (updated priiloader then to beta 3 update) and now when booting to the aroma forwarder, it shows the banner, then goes Black for a sec then goes back to the banner again, then im softlocked

**Reply by Ingunar** [reply]
> Update: I reinstalled both of them (updated priiloader then to beta 3 update) and now when booting to the aroma forwarder, it shows the banner, then goes Black for a sec then goes back to the banner again, then im softlocked

*(updated priiloader then to beta 3 update)* - I don't understand this, you didn't have RC3 installed? 

Did you download the `Priiloader_v0_10RC3.zip` and `PriiloaderWiiUForwarder.wuhb` from here? https://github.com/DacoTaco/priiloader/releases/tag/0.10.0-RC3

**Fully uninstall and install those files.**

**Reply by Siiyar** [reply]
I did. No difference.

**Reply by Ingunar** [reply]
I think that your Aroma might not be set up properly.

**Reply by Novek20** [reply]
> Priiloader forwarder for aroma doesn't seem to work at all. When launched, it just reboots the wii u

I am having the same problem

> I think that your Aroma might not be set up properly.

How do i set it up properly?
